enclave-runtime/src/test/direct_rpc_tests.rs

```diff
@@ -0,0 +1,78 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+	Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{rpc::worker_api_direct::public_api_rpc_handler, Hash};
+use codec::{Decode, Encode};
+use ita_stf::{Getter, TrustedGetter, TrustedGetterSigned};
+use itc_direct_rpc_server::{
+	create_determine_watch, rpc_connection_registry::ConnectionRegistry,
+	rpc_ws_handler::RpcWsHandler,
+};
+use itc_tls_websocket_server::{ConnectionToken, WebSocketMessageHandler};
+use itp_rpc::{RpcRequest, RpcReturnValue};
+use itp_stf_executor::{getter_executor::GetterExecutor, mocks::GetStateMock};
+use itp_stf_state_observer::mock::ObserveStateMock;
+use itp_top_pool_author::mocks::AuthorApiMock;
+use itp_types::{AccountId, DirectRequestStatus, Request, ShardIdentifier};
+use itp_utils::{FromHexPrefixed, ToHexPrefixed};
+use sp_core::ed25519::Signature;
+use sp_runtime::MultiSignature;
+use std::{string::ToString, sync::Arc, vec::Vec};
+
+pub fn get_state_request_works() {
+	type TestState = u64;
+
+	let connection_registry = Arc::new(ConnectionRegistry::<Hash, ConnectionToken>::new());
+	let watch_extractor = Arc::new(create_determine_watch::<Hash>());
+
+	let state: TestState = 78234u64;
+	let state_observer = Arc::new(ObserveStateMock::<TestState>::new(state));
+	let getter_executor =
+		Arc::new(GetterExecutor::<_, GetStateMock<TestState>>::new(state_observer));
+	let top_pool_author = Arc::new(AuthorApiMock::default());
+	let io_handler = public_api_rpc_handler(top_pool_author, getter_executor);
+	let rpc_handler = Arc::new(RpcWsHandler::new(io_handler, watch_extractor, connection_registry));
+
+	let getter = Getter::trusted(TrustedGetterSigned::new(
+		TrustedGetter::nonce(AccountId::new([0u8; 32])),
+		MultiSignature::Ed25519(Signature::from_raw([0u8; 64])),
+	));
+
+	let request = Request { shard: ShardIdentifier::default(), cyphertext: getter.encode() };
+
+	let request_string =
+		RpcRequest::compose_jsonrpc_call("state_get".to_string(), vec![request.to_hex()]).unwrap();
+
+	let response_string =
+		rpc_handler.handle_message(ConnectionToken(1), request_string).unwrap().unwrap();
+
+	assert!(!response_string.is_empty());
+
+	// Because we cannot de-serialize the RpcResponse here (unresolved serde_json and std/sgx feature issue),
+	// we hard-code the expected response.
+	//error!("{}", response_string);
```

Comment:
Is this code bit still up to date?

Reply:
This part of the code is brand new, and I wasn't able to solve the problem (spent about 1h and @clangenb 's time, but still coulnd't find a solution).

Reply:
I meant the error log :) But it's fine. You can leave as is

Reply:
I left the error log in, to update the hard-coded response string, if necessary.

Reply:
Ah interesting, the problem does only exist in the tests here? hmm...

Reply:
That problem exists anywhere inside the enclave (with feature `sgx`). We didn't notice it so far because we don't de-serialize `RpcResponse` in the enclave. But in tests it would be nice, as we can see here..